eth/sync.go

```diff
@@ -202,8 +202,17 @@ func (pm *ProtocolManager) synchronise(peer *peer) {
 		log.Info("Fast sync complete, auto disabling")
 		atomic.StoreUint32(&pm.fastSync, 0)
 	}
-	atomic.StoreUint32(&pm.acceptTxs, 1) // Mark initial sync done
-	if head := pm.blockchain.CurrentBlock(); head.NumberU64() > 0 {
+	// If we've successfully finished a sync cycle and passed any required checkpoint,
+	// enable accepting transactions from the network.
+	head := pm.blockchain.CurrentBlock()
+	if head.NumberU64() >= pm.checkpointNumber {
+		// Checkpoint passed, sanity check the timestamp to have a fallback mechanism
+		// for non-checkpointed (number = 0) private networks.
+		if head.Time() >= uint64(time.Now().Add(-30*24*time.Hour).Unix()) {
```

Comment:
Nitpick, why `Add` a negative number? Just makes the code less readable, at first glance it looks like you're doing it wrong

Reply:
Because there's no Sub :P

Reply:
Though we could use AddDate(https://golang.org/pkg/time/#Time.AddDate), might be a bit easier on the eye than -30*24*time.Hour.

Reply:
Switched it to AddDate, cleaner indeed.